### PR TITLE
Disable publish MenuItem

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose
Link to YouTrack:
[MAGN-7832](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7832#) Disable publish MenuItem when `IAuthProvider` is null or current workspace is not a `HomeWorkspaceModel`

Problem: extension is not notified by Dynamo, that current workspace changed. For this purpose created `Updated` action. `Updated` takes `ViewLoadedParams`. I could create something like `ViewUpdatedParams`, but then `ViewUpdatedParams` should be just inherited from `ReadyParams`. Because there is no any difference between `ReadyParams` and `ViewUpdatedParams`. They both should store current workspace and other opened workspaces.

### Declarations

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@pboyer

### CODE REVIEW

**Comment by pboyer** [comment]
Instead of adding the `Updated` method to `ViewExtension`, let's add an event to `ReadyParams`:

`event Action<IWorkspaceModel> WorkspaceChanged`

This event will have to respond to a `PropertyChanged` event on `DynamoModel` called "CurrentWorkspace", so it is slightly different.

**Comment by aosyatnik** [comment]
@pboyer , I've added `WorkspaceChanged` event. And test for it. Please, take a look. Thanks!

File: test/DynamoCoreWpfTests/ViewExtensionTests.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+using System.Windows;
+
+namespace DynamoCoreWpfTests
+{
+    public class ViewExtensionTests : DynamoTestUIBase
+    {
+        private DummyViewExtension viewExtension = new DummyViewExtension();
+
+        [Test]
+        public void OnWorspaceChangedExtensionIsNotified()
```
Comment (pboyer) [inline test/DynamoCoreWpfTests/ViewExtensionTests.cs]: Typo

File: src/DynamoCore/Extensions/ReadyParams.cs [hunk 2]
```diff
@@ -35,5 +37,18 @@ public IWorkspaceModel CurrentWorkspaceModel
                 return dynamoModel.CurrentWorkspace;
             }
         }
+
+        public event Action<IWorkspaceModel> WorkspaceChanged;
```
Comment (pboyer) [inline src/DynamoCore/Extensions/ReadyParams.cs]: Ok, just to be consistent, let's call this `OnCurrentWorkspaceModelChanged`.

**Comment by pboyer** [comment]
Thanks @aosyatnik! Just some minor comments and we're good.

**Comment by aosyatnik** [comment]
Hi, @pboyer , I merged master branch. PTAL, thanks :)

**Comment by pboyer** [comment]
Cool, thanks @aosyatnik!

**Pull request merged**